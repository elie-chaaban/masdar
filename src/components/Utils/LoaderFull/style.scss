@import '../../../assets/styles/fonts';
.loader-container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  position: fixed;
  left: 0;
  top: 0;
  font-family: $third-font, sans-serif;
  height: 100vh;
  z-index: 100;
  transition: all 0.5s;
  background: #404040;
  .lds-ellipsis {
    display: inline-block;
    position: relative;
    width: 80px;
    height: 15px;
  }
  .lds-ellipsis div {
    position: absolute;
    width: 13px;
    height: 13px;
    border-radius: 50%;
    animation-timing-function: cubic-bezier(0, 1, 1, 0);
  }
  .lds-ellipsis div:nth-child(1) {
    left: 8px;
    animation: lds-ellipsis1 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(2) {
    left: 8px;
    animation: lds-ellipsis2 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(3) {
    left: 32px;
    animation: lds-ellipsis2 0.6s infinite;
  }
  .lds-ellipsis div:nth-child(4) {
    left: 56px;
    animation: lds-ellipsis3 0.6s infinite;
  }
  .mobileLoader {
    font-size: 12px;
  }
  @keyframes lds-ellipsis1 {
    0% {
      transform: scale(0);
    }
    100% {
      transform: scale(1);
    }
  }
  @keyframes lds-ellipsis3 {
    0% {
      transform: scale(1);
    }
    100% {
      transform: scale(0);
    }
  }
  @keyframes lds-ellipsis2 {
    0% {
      transform: translate(0, 0);
    }
    100% {
      transform: translate(24px, 0);
    }
  }
  p {
    font-size: 1.5rem;
    text-transform: uppercase;
    color: white;
  }
}
