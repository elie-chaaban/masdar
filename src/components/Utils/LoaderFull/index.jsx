import React from 'react';
import {useSelector} from 'react-redux';
import './style.scss';

const LoaderFull = ({color = 'white'}) => {
  const {mobileMode} = useSelector((s: any) => s.styling);
  return (
    <div className="loader-container">
      <div className="text-center">
        <div className="lds-ellipsis">
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
        </div>
        {mobileMode ? <p className="mobileLoader">Loading...</p> : <p>Loading...</p>}
      </div>
    </div>
  );
};
export default LoaderFull;
