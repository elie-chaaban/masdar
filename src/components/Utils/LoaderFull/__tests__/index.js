import React from 'react';
import {shallow} from 'enzyme';
import LoaderFull from '../';

describe('Full Page Loader', () => {
  it('should have default props as white', () => {
    const dom = shallow(<LoaderFull />);
    expect(dom.find('.lds-ellipsis').childAt(0).prop('style')).toEqual({backgroundColor: 'white'});
    expect(dom).toMatchSnapshot();
  });
  it('should have passed props', () => {
    const dom = shallow(<LoaderFull color="red" />);
    expect(dom.find('.lds-ellipsis').childAt(0).prop('style')).toEqual({backgroundColor: 'red'});
  });
});
