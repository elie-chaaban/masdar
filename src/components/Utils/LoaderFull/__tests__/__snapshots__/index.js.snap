// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Page Loader should have default props as white 1`] = `
<div
  className="loader-container"
>
  <div
    className="text-center"
  >
    <div
      className="lds-ellipsis"
    >
      <div
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </div>
    <p>
      Loading...
    </p>
  </div>
</div>
`;
